File: src/main/java/de/codecentric/springnewspaper/controller/UploadController.java
package de.codecentric.springnewspaper.controller;

import de.codecentric.springnewspaper.domain.Newspaper;
import de.codecentric.springnewspaper.gateway.NewspaperGateway;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UploadController {
    private final NewspaperGateway newspaperGateway;

    public UploadController(@Autowired NewspaperGateway newspaperGateway) {
        this.newspaperGateway = newspaperGateway;
    }

    @PostMapping("/upload")
    public Newspaper uploadNewspaper(@RequestBody Newspaper newspaper) {
        newspaperGateway.put(newspaper);
        return newspaper;
    }
}

File: src/main/kotlin/de/codecentric/springnewspaper/controller/UploadController.kt
package de.codecentric.springnewspaper.controller

import de.codecentric.springnewspaper.domain.Newspaper
import de.codecentric.springnewspaper.gateway.NewspaperGateway
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UploadController(val gateway: NewspaperGateway) {
    @PostMapping("/upload")
    fun uploadNewspaper(@RequestBody newspaper: Newspaper): Newspaper {
        gateway.put(newspaper)
        return newspaper
    }
}
